src/commands/environments/update.ts

```diff
@@ -10,33 +11,68 @@ import {
     EnvironmentPromptResult
 } from '../../ui/prompts'
 import UpdateCommand from '../updateCommand'
+import { Flags } from '@oclif/core'
 
-export default class UpdateEnvironment extends UpdateCommand<CreateEnvironmentParams> {
+export default class UpdateEnvironment extends UpdateCommand {
     static hidden = false
     static description = 'Update a Environment.'
 
     prompts = [
         namePrompt,
         descriptionPrompt
     ]
+    static flags = {
+        ...UpdateCommand.flags,
+        'type': Flags.string({
+            description: 'The type of environment',
+            options: environmentTypes
+        }),
+        'description': Flags.string({
+            description: 'Description for the environment',
+        }),
+    }
 
     public async run(): Promise<void> {
+        const { flags } = await this.parse(UpdateEnvironment)
+        const { key, name, type, description, headless } = flags
+
         await this.requireProject()
-        const { environment } = await inquirer.prompt<EnvironmentPromptResult>([environmentPrompt], {
-            token: this.authToken,
-            projectKey: this.projectKey
-        })
+        let envKey: string
+        if (headless) {
+            if (!key) {
+                this.writer.showError('In headless mode, the key flag is required')
```

Comment:
I know this was added in some other places, but I don't think "In headless mode" is necessary in this message. The key/ID is always required, it's just that in headless mode we don't prompt for it